src/derivative_wrappers.jl

```diff
@@ -56,15 +52,17 @@ function jacobian(f, x, integrator)
     J
 end
 
-jacobian_autodiff(f, x) = (ForwardDiff.derivative(f,x),1)
-jacobian_autodiff(f, x::AbstractArray) = (ForwardDiff.jacobian(f, x),1)
+jacobian_autodiff(f, x, integrator) = (ForwardDiff.derivative(f,x),1)
 function jacobian_autodiff(f, x::AbstractArray, integrator)
```

Comment:
Why is `integrator` passed as argument? It seems only `integrator.f` is accessed in the function body - shouldn't these calls be replaced by `f`, in which case `integrator` would be needless?

Reply:
I think it is just a legacy.

Reply:
I don't see how passing `integrator.f` instead of `integrator` addresses the point that I wanted to make? IMO there should be only two arguments to `jacobian_autodiff`.

Reply:
Because it is colored jacobian.

Reply:
Sorry, I don't quite understand. My question is: why can we not replace all occurrences of `odefun` by `f`?

Reply:
Oh, I thought `f` is the unwrapped function from `ODEFunction` object.

Reply:
Well, it turns out the `f` is indeed the unwrapped function. I replaced `odefun` with `f` and got ` type UDerivativeWrapper has no field jac_prototype`.

Reply:
Ah I see, thanks for trying! I'm wondering if it would make sense to add more traits and `getproperty` overloads to `UDerivativeWrapper` and `UJacobianWrapper` (as done in https://github.com/JuliaDiffEq/DiffEqBase.jl/blob/c65c0e7442b63332aac60002baa9ebb021890e76/src/diffeqfunction.jl#L883-L888) instead of passing `integrator.f`

Reply:
I guess that's not very friendly for extension unless there is a generic way to do it.